lib/subordinateio.c

```diff
@@ -738,7 +795,19 @@ struct subordinate_range **list_owner_ranges(const char *owner, enum subid_type
 	const struct subordinate_range *range;
 	struct subordinate_range **ranges = NULL;
 	struct commonio_db *db;
+	bool ok;
+	enum subid_status status;
 	int size = 0;
+	struct subid_nss_ops *h;
+
+	nss_init(NULL);
+	h = get_subid_nss_handle();
+	if (h) {
+		status = h->list_owner_ranges(owner, id_type, &ranges, &ok);
+		if (status == SUBID_STATUS_SUCCESS && ok)
+			return ranges;
+		return NULL;
+	}
```

Comment:
I think `ok` is excessive.
It could be just
```
    status = h->list_owner_ranges(owner, id_type, &ranges);
    if (status == SUBID_STATUS_SUCCESS)
        return ranges;
    return NULL;
```

Reply:
Hm, well I was thinking for the SUBID_STATUS_UNKNOWN_USER case - but i guess what i really wanted is satisfied by empty list or count=0.  So I think I'll drop that, thanks.

Reply:
Any `return code != SUBID_STATUS_SUCCESS` means caller shouldn't read out-args at all.

user not found => just `SUBID_STATUS_UNKNOWN_USER`
user found, but doesn't have any subid range assigned => SUBID_STATUS_SUCCESS && (*ranges == NULL)
user found and has ranges => SUBID_STATUS_SUCCESS && (*ranges != NULL)

imo that way make sense.

Reply:
Well, user unknown seemed special to me, but that's sensible.

Reply:
The same logic applies to all plugin functions.